### PR TITLE
Improve checking of changes in module or module.exports

### PR DESCRIPTION
**Pull request by Yeti-or**
check for changes in `module` or `module.exports`

### CODE REVIEW

**Comment by Yeti-or** [comment]
fix #4 
my longest commit msg ever, @zxqfox check it out

File: index.js [hunk 1]
```diff
@@ -45,7 +45,19 @@ module.exports = function(content, filename, context) {
 
     var result;
     if(sandbox.module.exports['__42__'] && Object.keys(sandbox.module.exports).length === 1) {
+        var needToClean;
+        if(global.module) {
+            needToClean = global.module;
```
Comment (qfox) [inline index.js]: NOOOOOOOOOOOOO

Comment (qfox) [inline index.js]: I'd just check module Object for new keys

Probably we shouldn't do this

Comment (Yeti-or) [inline index.js]: mmm, ye it's seems less error prone

**Comment by Yeti-or** [comment]
🆙

**Comment by qfox** [comment]
👍 :shipit:

File: index.js [hunk 2]
```diff
@@ -41,13 +41,17 @@ module.exports = function(content, filename, context) {
...
 
     var result;
-    if(sandbox.module.exports['__42__'] && Object.keys(sandbox.module.exports).length === 1) {
+    if(
+        Object.keys(sandbox.module.exports).length === exportKeysCount &&
+        Object.keys(sandbox.module).length === moduleKeysCount
+    ) {
         result = context ? vm.runInNewContext(content, context) : vm.runInThisContext(content);
     } else {
-        delete sandbox.module.exports['__42__'];
```
Comment (qfox) [inline index.js:50]: I'm looking for tests. I don't see where we checking case with pushing data to `exports` object.

Can you recheck it?

``` js
nodeEval('exports.lolka = 123').shouldEqual({lolka: 123});
```

Comment (Yeti-or) [inline index.js:50]:
```
213:node-eval yeti-or$ node
> var nn = require('.')
undefined
> nn('exports.lol = 123')
{ lol: 123 }
> nn('var t= 12; exports.lol = 123')
{ lol: 123 }
```

Comment (Yeti-or) [inline index.js:50]: https://github.com/gulp-bem/node-eval/blob/master/test/test.js#L25-L29

Comment (qfox) [inline index.js:50]: What about `module.exports = 42`?

Comment (qfox) [inline index.js:50]: Okkk

Comment (Yeti-or) [inline index.js:50]: shiiiit

**Pull request merged**